lib/Transforms/TypeConversion.cpp

```diff
@@ -0,0 +1,83 @@
+//===- TypeConversion.cpp -  --------------*- C++ -*-===//
```

Comment:
I suggest renaming the files 'TypeConversion*' into something less general. They are not about general (ar all) type-conversion(s). Moreover, why are they in 'Transforms'? They are useful for Conversions, too. Like above, I suggest to place them into a more generic place internal/utils.